File: 1014-best-sightseeing-pair/1014-best-sightseeing-pair.py
class Solution:
    def maxScoreSightseeingPair(self, values: List[int]) -> int:
        cur,res = 0,0
        for v in values:
            res = max(res, cur+v) # 결과는 현재냐 아니면 전에거 추가냐
            cur = max(cur, v) -1
            
        return res

File: 1014-best-sightseeing-pair/1014-best-sightseeing-pair.rs
impl Solution {
    pub fn max_score_sightseeing_pair(values: Vec<i32>) -> i32 {
        let mut res = 0;
        let mut cur = 0;

        for val in values {
            res = res.max(cur + val);
            cur = cur.max(val) - 1;
        }

        res
    }
}